### PR TITLE
Convert winlogbeat to package

### PR DESCRIPTION
**Pull request by leehinman**
## What does this PR do?

Adds Winlogbeat functionality as a package

- use filebeat winlog input
- include application, forwarded, powershell, powershell_operational
  security, sysmon_operational and system datasets
- converted winlogbeat visualizations

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md#tips-for-building-integrations) and this pull request is aligned with them.
- [x] I have verified that all datasets collect metrics or logs.

## How to test this PR locally

 - sysmon needs to be installed
 - recent filebeat 8.0.0 needs to be used by agent

## Screenshots

win_pkg_1
win_pkg_2
win_pkg_3
win_pkg_4
win_pkg_5
win_pkg_6
win_pkg_7

### CODE REVIEW

**Comment by jamiehynds** [comment]
@leehinman Great to see the PR! Do you think we could rework the 'Collect events from Windows event log' section? As we add additional Windows event modules, I fear that section might become unwieldy, with lots of duplicated text. 

Maybe we could do 'Collect events from the following Windows event log channels:

- Application
- System 
- Security
- PowerShell
- Windows Forwarded Events
- etc....

File: packages/windows/dataset/powershell/fields/winlog.yml [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+          type: keyword
+    - name: event_id
+      type: keyword
+      required: true
+      description: >
+        The event identifier. The value is specific to the source of the event.
+
+    - name: keywords
+      type: keyword
+      required: false
```
Comment (andrewkroh) [inline packages/windows/dataset/powershell/fields/winlog.yml:270]: These `required` fields came over from Beats, but are they part of the package spec? They were never used for anything in Beats AFAIK. We can remove them later.

**Review by andrewkroh — APPROVED** [review]
Awesome work! 👏

**Comment by ruflin** [comment]
@leehinman This is great!
@andresrc @mukeshelastic @fearful-symmetry For awareness, here is a Windows package in the works!

**Comment by leehinman** [comment]
> @leehinman Great to see the PR! Do you think we could rework the 'Collect events from Windows event log' section? As we add additional Windows event modules, I fear that section might become unwieldy, with lots of duplicated text.
> 
> Maybe we could do 'Collect events from the following Windows event log channels:
> 
> * Application
> * System
> * Security
> * PowerShell
> * Windows Forwarded Events
> * etc....

@jamiehynds thanks for the suggestion.  I came up with 2 ideas.

1) Use official channel name only, but this doesn't look alphabetical because it is sorted by dataset name:
win_pkg_8

or 

2) Use dataset name, add official channel name in description if different from dataset name:
win_pkg_9

What do you think?

File: packages/windows/dataset/powershell/fields/winlog.yml (hunk 1 above)
Comment (leehinman) [reply packages/windows/dataset/powershell/fields/winlog.yml:270]: I'll investigate.

**Comment by andrewkroh** [comment]
I think I like the first one. (Though I am surprised how quickly the text begins to wrap around.)

**Comment by andrewkroh** [comment]
@mtojek mentioned that event.module is no longer populated. The windows event packages still set that field via their script processors. IIUC we need to update this to set `event.dataset` to a value like `windows.sysmon` and remove `event.module`.

**Comment by jamiehynds** [comment]
@leehinman I'd vote for the second one, mainly because it's in alphabetical order so easier for users to find a dataset and they can also reference the official channel name if needed. 

On a different note, what's the default experience for users? Are all Windows datasets enabled by default?

**Comment by MikePaquette** [comment]
@andrewkroh 

>  IIUC we need to update this to set event.dataset to a value like windows.sysmon and remove event.module.

Why are we removing `event.module`? For BWC, can we continue to populate `event.module` as well?

**Comment by leehinman** [comment]
> @leehinman I'd vote for the second one, mainly because it's in alphabetical order so easier for users to find a dataset and they can also reference the official channel name if needed.

Thanks, I'll merge the 2nd version.


> On a different note, what's the default experience for users? Are all Windows datasets enabled by default?

Yes, all of them are enabled by default (metrics & logs and all datasets in them).

**Comment by leehinman** [comment]
> @mtojek mentioned that event.module is no longer populated for filebeat log packages. The windows event packages still set that field via their script processors. IIUC we need to update this to set `event.dataset` to a value like `windows.sysmon` and remove `event.module`.
> 
> If you equate a package to a module then the `event.module` values we had in Winlogbeat don't make sense (powershell, security, sysmon). The module value would need to be "windows". If none of the other log packages (converted from Filebeat) are setting `event.module` then this seems inconsistent to continue setting it.

`event.dataset` is already being set (eg. windows.security).  You ok if we merge this now and fix up `event.module` here and in Winlogbeat at the same time?

**Comment by andrewkroh** [comment]
> event.dataset is already being set (eg. windows.security). You ok if we merge this now and fix up event.module here and in Winlogbeat at the same time?

Yes, we can fix up `event.module` later. I don't think there's anything to change within Winlogbeat, just here.

**Pull request merged**